### DISCUSSION TITLE
Adia for design 'Too many endpoint' for Hono

### DISCUSSION THREAD

**Discussion by roccho-dev** [honojs/hono#2957] [Ideas]
Does the following code neutralize Hono's characteristic RegExp search process?

```
const routes = [
  { method: 'PURGE', path: '/cache', handler: (c) => c.text('PURGE Method /cache') },
  { method: ['PUT', 'DELETE'], path: '/post', handler: (c) => c.text('PUT or DELETE /post') },
  { method: 'GET', path: ['/hello', '/ja/hello', '/en/hello'], handler: (c) => c.text('Hello') }
];

routes.forEach(route => {
  app.on(route.method, route.path, route.handler);
});
```

**Comment by roccho-dev** [comment]
OR

Is the performance impact minimal since the definition of routes by this code is usually done only once at application startup?

**Comment by usualoma** [comment]
Hi @PorcoRosso85 

### Short answer

The code you have shown will not degrade performance.

### Long answer

Having more endpoints does not degrade performance compared to other routers. I have confirmed no performance degradation with a few hundred or so endpoints.

Due to the optimization process during startup, it also performs very well when a large number of middleware is combined.

As an internal implementation, if there is no path parameter, as in your example, it is looked up from the object. This case is very fast.
```ts
{
  '/hello' => handler,.
  '/en/hello' => handler,.
  '/en/hello' => handler,.
}[requestPath]
```
